app/javascript/components/DownloadButton.js

```diff
@@ -115,6 +145,59 @@ function DownloadCommandContainer(props) {
   )
 }
 
+/**
+ * Format number in bytes, with human-friendly units
+ *
+ * Derived from https://gist.github.com/lanqy/5193417#gistcomment-2663632
+ */
+function bytesToSize(bytes) {
```

Comment:
Would it help to have a summary number in the API response to this so you don't have to do the math client side?  Just a thought.